src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/PartiallyDeterminedHaplotypeComputationEngine.java

```diff
@@ -249,21 +188,12 @@ We iterate over every ref position and select single alleles (including ref) fro
             }
         }
 
-        if (outputHaplotypes.size() > MAX_PD_HAPS_TO_GENERATE) {
-            Utils.printIf(debug,() -> "Too many branch haplotypes found, aborting ["+outputHaplotypes.size()+"]");
-            return sourceSet;
-        }
+        // TODO: Sorting haplotypes is unnecessary but makes debugging against previous versions much easier.
+        final List<Haplotype> result = outputHaplotypes.stream().sorted(Comparator.comparing(Haplotype::getBaseString)).toList();
         sourceSet.storeAssemblyHaplotypes();
-
-        // TODO this is an entirely unnecessary step that can be done away with but i leave in because it makes debugging against previous versions much easier.
-        final Set<Haplotype> result = outputHaplotypes.stream().sorted(Comparator.comparing(Haplotype::getBaseString)).collect(Collectors.toCollection(LinkedHashSet::new));
         sourceSet.replaceAllHaplotypes(result);
-        Utils.printIf(debug, () -> "Constructed Haps for Branch"+sourceSet.getHaplotypeList().stream().map(Haplotype::toString).collect(Collectors.joining("\n")));
-        if (!pileupArgs.determinePDHaps) {
-            // Setting a boolean on the source-set to indicate to downstream code that we have PD haplotypes
-            sourceSet.setPartiallyDeterminedMode();
-        }
-        Utils.printIf(debug, () -> "Returning "+outputHaplotypes.size()+" to the HMM");
+        sourceSet.setPartiallyDeterminedMode(!pileupArgs.determinePDHaps);
```

Comment:
add a comment that this is necessary to flag later PDHMM genotyping (and that we want to set it at the end here because the above code might fail out and thus this is a bell weather)